WordPress/openverse
# Display generated tags separately

## Fixes
Fixes #4192 by @fcoveram 

## Description
This PR adds a separate display of machine-generated tags and tags provided by sources on the single result page following the [Figma mockups](https://www.figma.com/file/BUZ0VjjzGigUuICDlhEttG/Machine-generated-tags?type=design&node-id=22-13656&mode=dev).

Based on @fcoveram's [suggestion](https://github.com/WordPress/openverse/pull/4291#issuecomment-2136714826), if the item only has source tags, the tags section doesn't have a visible heading.

~~I started to write the implementation plan for the frontend feature, but realized that this can be done in a single PR. As the project lead, do you think we need a frontend IP, @AetherUnbound?~~

This PR is blocked on #4379 because it has a link to the "About tags" page, but the page is not ready.
Another blocker is the text for this link. Currently, the link says "What is this?"; suggestions for the text are welcome.

~~The PR can be reviewed but should not be merged before #4379 is done.~~
#4379 was merged into this PR. Please also review the wording here.

## Testing instructions
Run the app using `just frontend/run dev` (running `dev:only` would not update the English strings, so you would need to run `dev` at least once)
Go to http://localhost:8443/image/a487f4eb-ce05-43e1-acae-73c4ab090cc9 to see how machine-generated and source tags are displayed.
